api/model/src/main/java/org/projectnessie/model/RepositoryInformation.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonView;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.time.Instant;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.immutables.value.Value;
+import org.projectnessie.model.CommitMeta.InstantDeserializer;
+import org.projectnessie.model.CommitMeta.InstantSerializer;
+import org.projectnessie.model.ser.Views;
+
+/**
+ * Informational object to tell a client about the Nessie repository.
+ *
+ * <p>All values are optional and may or may not be present, depending on the actual server version.
+ */
+@Schema(
+    type = SchemaType.OBJECT,
+    title = "RepositoryInformation",
+    description =
+        "Exposes some information about the Nessie repository.\n\n"
+            + "All attributes may or may not be present, depending on the server version and repository.")
+@Value.Immutable
+@JsonSerialize(as = ImmutableRepositoryInformation.class)
+@JsonDeserialize(as = ImmutableRepositoryInformation.class)
+@JsonView(Views.V2.class)
+public interface RepositoryInformation {
+  @Nullable
+  @jakarta.annotation.Nullable
+  String getDefaultBranch();
```

Comment:
If we intend repo info to be the source of truth for the default branch name, should we deprecate it in `NessieConfiguration` perhaps?